Scenes/Enemies/TemplateEnemy/TemplateEnemy.tscn

```diff
@@ -1,7 +1,25 @@
-[gd_scene load_steps=2 format=2]
+[gd_scene load_steps=4 format=2]
 
 [ext_resource path="res://Scenes/Enemies/TemplateEnemy/TemplateEnemy.gd" type="Script" id=1]
 
+[sub_resource type="Shader" id=1]
+code = "shader_type canvas_item;
+uniform bool turn_red;
+
+void fragment() {
+	vec4 color = texture(TEXTURE, UV);
+	if (turn_red) {
+		COLOR = vec4(1, 0, 0, 1);
+	} else {
+		COLOR = color;
+	}
+}"
```

Comment:
weirrddddddddd I didn't realize it was writing code in these files

Reply:
I DID THAT!! when you make a new shader you write code for it because is a separate program that runs on the GPU. super cool

Reply:
That is actually super cool